src/Rules/PHPUnit/PHPUnitVersionDetector.php

```
@@ -13,33 +14,33 @@
 class PHPUnitVersionDetector
 {
 
-	private ReflectionProvider $reflectionProvider;
-
-	public function __construct(ReflectionProvider $reflectionProvider)
-	{
-		$this->reflectionProvider = $reflectionProvider;
-	}
-
 	public function createPHPUnitVersion(): PHPUnitVersion
 	{
+		$file = false;
 		$majorVersion = null;
 		$minorVersion = null;
-		if ($this->reflectionProvider->hasClass(TestCase::class)) {
-			$testCase = $this->reflectionProvider->getClass(TestCase::class);
-			$file = $testCase->getFileName();
-			if ($file !== null) {
-				$phpUnitRoot = dirname($file, 3);
-				$phpUnitComposer = $phpUnitRoot . '/composer.json';
-				if (is_file($phpUnitComposer)) {
-					$composerJson = @file_get_contents($phpUnitComposer);
-					if ($composerJson !== false) {
-						$json = json_decode($composerJson, true);
-						$version = $json['extra']['branch-alias']['dev-main'] ?? null;
-						if ($version !== null) {
-							$versionParts = explode('.', $version);
-							$majorVersion = (int) $versionParts[0];
-							$minorVersion = (int) $versionParts[1];
-						}
+
+		try {
+			// uses runtime reflection to reduce unnecessary work while bootstrapping PHPStan.
+			// static reflection would need to AST parse and build up reflection for a lot of files otherwise.
+			$reflection = new ReflectionClass(TestCase::class);
+			$file = $reflection->getFileName();
+		} catch (ReflectionException $e) {
+			// PHPUnit might not be installed
+		}
+
+		if ($file !== false) {
+			$phpUnitRoot = dirname($file, 3);
+			$phpUnitComposer = $phpUnitRoot . '/composer.json';
+			if (is_file($phpUnitComposer)) {
```

do we really need this `is_file` or could we just drop it, because we already do `@file_get_contents($phpUnitComposer);` which would swallow the error nevertheless?

---

Could be dropped :)

---

here we go.